eth/api_debug.go

```diff
@@ -173,8 +176,11 @@ type storageEntry struct {
 
 // StorageRangeAt returns the storage at the given block height and transaction index.
 func (api *DebugAPI) StorageRangeAt(ctx context.Context, blockNrOrHash rpc.BlockNumberOrHash, txIndex int, contractAddress common.Address, keyStart hexutil.Bytes, maxResult int) (StorageRangeResult, error) {
-	var block *types.Block
+	if api.eth.blockchain.CacheConfig().StateScheme == customrawdb.FirewoodScheme {
+		return StorageRangeResult{}, errFirewoodNotSupported
+	}
 
+	var block *types.Block
 	block, err := api.eth.APIBackend.BlockByNumberOrHash(ctx, blockNrOrHash)
```

Comment:
Is adding the additional block declaration necessary? I know `err` has to be declared and assigned to but I've not seen this pattern before.